Gadget SDK - Keystore Module

shekohex opened the pull request:
Closes #155

Tjemmmic commented on sdk/src/keystore/backend/mem.rs:
```diff
@@ -0,0 +1,274 @@
+//! In Memory Keystore Backend that supports different cryptographic key operations such as key generation, signing, and public key retrieval.
```
Not pressing, but `In Memory` generally uses a hyphen to avoid confusion when reading: `In-Memory`

Tjemmmic commented on sdk/src/keystore/backend/mem.rs:
```diff
@@ -0,0 +1,274 @@
+    }
+}
+
+/// The In Memory Keystore Backend. It stores everything in memory
```
`In-Memory` again, not sure if it is a big deal. I just think it helps with readability.

Tjemmmic commented on sdk/src/keystore/backend/mod.rs:
```diff
@@ -0,0 +1,8 @@
+//! Keystore backend implementations.
+
+/// In Memory Keystore Backend
```
`In-Memory`

Tjemmmic commented on sdk/src/keystore/mod.rs:
```diff
@@ -0,0 +1,186 @@
+//! - `memory`: A simple in-memory storage (enabled by default)
+//! - `file`: A storage that saves the keys to a file (`keystore-file` feature)
+//! - `kms`: A storage that uses a AWS Key Management System (`keystore-kms` feature)
+//! - `substrate`: Uses Substrate's Framewrok keystore (`keystore-substrate` feature)
```
Typo: Framework

Tjemmmic commented on sdk/src/keystore/error.rs:
```diff
@@ -0,0 +1,35 @@
+    #[cfg(feature = "keystore-ecdsa")]
+    #[error("ecdsa error: {0}")]
+    Ecdsa(#[from] k256::elliptic_curve::Error),
+
```
Formatting: Inconsistent whitespace

Tjemmmic commented on sdk/src/keystore/mod.rs:
```diff
@@ -0,0 +1,186 @@
+/// ECDSA Support
+#[cfg(feature = "keystore-ecdsa")]
+mod ecdsa;
+/// Ed25519 Support
```
Formatting: Inconsistent Whitespace

Tjemmmic commented on sdk/src/keystore/mod.rs:
```diff
@@ -0,0 +1,186 @@
+        &self,
+        public: &sr25519::Public,
+    ) -> Result<Option<sr25519::Secret>, Error>;
+
```
Formatting: Inconsistent Whitespace

Tjemmmic commented on sdk/src/random.rs:
```diff
@@ -0,0 +1,45 @@
+pub fn getrandom_or_panic() -> impl RngCore + CryptoRng {
+    const PRM: &'static str = "Attempted to use functionality that requires system randomness!!";
+
+    // Should we panic when invoked or when used?
```
Possible leftover comment or TODO

Tjemmmic commented on sdk/src/keystore/backend/fs.rs:
```diff
@@ -0,0 +1,270 @@
+    Bls381 = 0x03,
+}
+
+/// The filesystem keystore backend. it stores keys in files, where each file
```
Capitalization of "it" - random grammar, not vital

Tjemmmic reviewed (commented):
Mostly just things related to comments, readability, etc. Nothing super important or related to actual code

shekohex commented:
> Mostly just things related to comments, readability, etc. Nothing super important or related to actual code

Thank you, addressed all of your comments.
